### PR TITLE
add tool skeleton.py

### PR DESCRIPTION
**Pull request by starius**

### CODE REVIEW

**Comment by tonytheodore** [comment]
Nice! I like this idea and it's a good way to start rationalising future packages.

In the short term, I'd propose adding [`$(SOURCE_DIR)` and `$(BUILD_DIR)`](https://github.com/mxe/mxe/pull/1193/commits/65d0fab27a7873e10360afb2888aed29a4fbc874) and changing `$(2).c` to `$(PKG)-test.c` to remove all positional variable references.

Ideally, I'd also like to [remove metadata from `index.html`](https://github.com/mxe/mxe/issues/1422) but that shouldn't hold this up.

**Comment by tonytheodore** [comment]
I've [added an Outline section](https://github.com/mxe/mxe/wiki/Customisable-Packages/_compare/1da8a226768c6fe086192c828a32230bcf5d0916...046d378a37e972300192805a4229c7497d400a14) to the wiki https://github.com/mxe/mxe/wiki/Customisable-Packages#package-makefiles

**Comment by starius** [comment]
@tonytheodore, I reapplied https://github.com/mxe/mxe/pull/1193/commits/65d0fab27a7873e10360afb2888aed29a4fbc874 and used `$(SOURCE_DIR)` and `$(BUILD_DIR)` in `skeleton.py`.

I do not understand how to replace `$(2).c` with `$(PKG)-test.c`, because `$(2)` is expanded to `$(TOP_DIR)/src/$(1)-test`.

**Comment by tonytheodore** [comment]
> because $(2) is expanded to $(TOP_DIR)/src/$(1)-test

Of course! sorry for the confusion. Maybe `$(TEST_FILE)` instead - https://github.com/LuaAndC/mxe/pull/8.

File: tools/skeleton.py [hunk 1]
```diff
@@ -40,29 +40,27 @@
...
         noinst_PROGRAMS=
 '''
 
 MAKE_BUILD = r'''
     # build and install the library
-    $(MAKE) -C '$(1)' -j '$(JOBS)'
-    $(MAKE) -C '$(1)' -j 1 install
+    $(MAKE) -C '$(SOURCE_DIR)' -j '$(JOBS)'
+    $(MAKE) -C '$(SOURCE_DIR)' -j 1 install
 '''
```
Comment (tonytheodore) [inline tools/skeleton.py]: s/SOURCE_DIR/BUILD_DIR

**Comment by starius** [comment]
Thanks! I merged pull request https://github.com/LuaAndC/mxe/pull/8 and [used](https://github.com/mxe/mxe/pull/1432/commits/56ab126366877178f5cfbc992ac1867ce0c7dbe7) `$(TEST_FILE)` in skeleton.py

**Comment by starius** [comment]
I forgot to test with Python 3. Fixed.

**Comment by tonytheodore** [comment]
Thanks!

**Pull request merged**